sljit_src/sljitNativeARM_64.c

```diff
@@ -253,8 +255,10 @@ SLJIT_API_FUNC_ATTRIBUTE void* sljit_generate_code(struct sljit_compiler *compil
 	CHECK_PTR(check_sljit_generate_code(compiler));
 	reverse_buf(compiler);
 
+	SLJIT_SCOPE_ENABLE_JIT_WRITE();
 	code = (sljit_ins*)SLJIT_MALLOC_EXEC(compiler->size * sizeof(sljit_ins));
 	PTR_FAIL_WITH_EXEC_IF(code);
```

Comment:
`PTR_FAIL_WITH_EXEC_IF` does early return. This might affect the enabling logic.

Reply:
The macros use a `__attribute__(cleanup(<function>))` to ensure state is restore at all return points.

Reply:
while the use of the cleanup attribute might had been useful where this code was lifted from, as shown in #52, there are only a few places (even less if the allocator is changed first, so it doesn't keep the chunk headers together with the code) where this need to be added and having it done explicitly makes the code simpler and easier to follow